### PR TITLE
Initialize Interpolation with Float and Double types

### PR DESCRIPTION
**Pull request by AliceAponasko**
https://github.com/square/Paralayout/issues/26

### CODE REVIEW

**Comment by NickEntin** [comment]
Looks great! Let's retarget against `develop/1.0` so this lands in the 1.0 release and maybe worth adding some tests for these? There's an `InterpolationTests` class on that branch we can add test cases to.

File: ParalayoutTests/InterpolationTests.swift [hunk 1]
```diff
@@ -29,4 +31,248 @@ final class InterpolationTests: XCTestCase {
         XCTAssertEqual(actualRect, expectedRect)
     }
 
+    // MARK: - Float
+
+    func testLinearFloatInterpolation_fromMinToMax() {
```
Comment (AliceAponasko) [inline ParalayoutTests/InterpolationTests.swift:36]: @NickEntin I've added some tests, these cover base cases, but do not cover every curve combination possible (especially for the function with a midpoint value). Let me know if this is good enough!

File: ParalayoutTests/InterpolationTests.swift [hunk 2]
```diff
@@ -29,4 +31,248 @@ final class InterpolationTests: XCTestCase {
...
+    }
+
+    func testEaseInFloatInterpolation_fromMinToMax() {
+        let value: Float = 3.0
+        let min: Float = 1.0
+        let max: Float = 5.0
+        let interpolation = Interpolation(of: value, from: min, to: max)
+            .interpolate(from: min, to: max, curve: .easeIn)
+
+        XCTAssertEqual(interpolation, 2)
```
Comment (NickEntin) [inline ParalayoutTests/InterpolationTests.swift:53]: I suspect we may need to add an `accuracy` parameter here to account for slight rounding errors across machines (for all of these assertions).

File: ParalayoutTests/InterpolationTests.swift [hunk 3]
```diff
@@ -29,4 +31,248 @@ final class InterpolationTests: XCTestCase {
...
+        let value: Double = 3.0
+        let min: Double = 1.0
+        let max: Double = 5.0
+        let interpolation = Interpolation(of: value, from: min, to: max)
+            .interpolate(from: min, to: max)
+
+        XCTAssertEqual(interpolation, 3)
+    }
+
+    func testEaseInDoubleDoublepolation_fromMinToMax() {
```
Comment (NickEntin) [inline ParalayoutTests/InterpolationTests.swift:168]:
```suggestion
    func testEaseInDoubleInterpolation_fromMinToMax() {
```

File: ParalayoutTests/InterpolationTests.swift (hunk 1 above)
Comment (NickEntin) [reply ParalayoutTests/InterpolationTests.swift:36]: These look fantastic! 🙌

**Review by NickEntin — APPROVED** [review]

File: ParalayoutTests/InterpolationTests.swift (hunk 2 above)
Comment (AliceAponasko) [reply ParalayoutTests/InterpolationTests.swift:53]: Hm, I'm not sure what do you mean, would you mind elaborating? I haven't seen any rounding issues, the numbers in this calculation add up to `2`.

Comment (AliceAponasko) [reply ParalayoutTests/InterpolationTests.swift:53]: Hey @NickEntin! I'm going to merge, but if you have any suggestions or concerns, I can address those in my next PRs!

Comment (NickEntin) [reply ParalayoutTests/InterpolationTests.swift:53]: Sorry, forgot to respond to that one. As long as the builds are passing, I think it's fine for now. 👍

**Pull request merged**